mlflow/projects/__init__.py

```diff
@@ -91,25 +68,25 @@ def _run(uri, experiment_id, entry_point, version, parameters,
     Returns a ``SubmittedRun`` corresponding to the project run.
     """
     tracking_store_uri = tracking.get_tracking_uri()
+    backend_config[PROJECT_USE_CONDA] = use_conda
+    backend_config[PROJECT_SYNCHRONOUS] = synchronous
+    backend_config[PROJECT_DOCKER_ARGS] = docker_args
+    backend_config[PROJECT_STORAGE_DIR] = storage_dir
     # TODO: remove this check once local, databricks, kubernetes execution have been refactored
     # into their own built-in project execution backends.
-    if backend_name not in {"local", "databricks", "kubernetes"}:
+    if backend_name not in {"databricks", "kubernetes"}:
```

Comment:
Question: shouldn't we remove kubernetes from this list? Given that we've refactored it into its own backend in this PR

Reply:
Unfortunate (but not the fault of this PR) that tests didn't catch this - there seems to be no test that calls `mlflow.run` with `backend="databricks"`, could we add one to this PR? We might just be able to modify one of the existing tests which already mocks out all the actual k8s job-submission logic, e.g. [this one](https://github.com/mlflow/mlflow/blob/f291f5bc8230cb44cfe25748505d3189dafe8aaf/tests/projects/test_kubernetes.py#L89) that directly calls the `run` method of the `KubernetesBackend`, and simply call `mlflow.projects.run` instead to more-closely verify things work end-to-end.

Reply:
Agree. I will add one in the PR related to kubernetes